sympy/physics/quantum/density.py

```diff
@@ -168,6 +170,10 @@ def doit(self, **hints):
                 for arg in product(state.args, repeat=2):
                     terms.append(prob *
                                  self._generate_outer_prod(arg[0], arg[1]))
+            elif state!=null:
+                terms.append(prob *
+                                 self._generate_outer_prod(arg[0]))
```

Comment:
`null` is not a valid Python keyword (perhaps `None` was intended) and `_generate_outer_prod` requirements two arguments, so this part of the PR seems syntactically and semantically invalid.